src/include/migraphx/compile_options.hpp

```diff
@@ -40,6 +41,8 @@ struct compile_options
 
     bool fast_math       = true;
     bool exhaustive_tune = false;
+    bool weight_streaming = false;
+    long max_memory       = LONG_MAX;
```

Comment:
Actually thinking about this more, I think we want it set to `0` for default, which would mean to use the memory available on the gpu, and then the user can set max to always disable weight streaming.

Also, this should probably be `std::size_t` type.

Reply:
What would be the point of setting the max if it always disables weight streaming? I understand the idea of 0 for default (aka 0 = make it run no matter what), but if setting the max disables weight streaming than it would either a. always fail with not enough memory, or b. always work since the max is > size of literals.